datalad/support/archives.py

```diff
@@ -11,131 +11,53 @@
-#
-# def _patool_log_info(msg, *args, **kwargs):
-#     _patool_log(logging.DEBUG, msg)
-#
-# def _patool_log_error(msg, *args, **kwargs):
-#     _patool_log(logging.ERROR, msg)
-#
-# patoolib.util.log_info = _patool_log_info
-# patoolib.util.log_error = _patool_log_error
-# patoolib.util.log_internal_error = _patool_log_error
-
-# we need to decorate patool.util.run
-# because otherwise it just lets processes to spit out everything to std and we
-# do want to use it at "verbosity>=0" so we could get idea on what is going on.
-# And I don't want to mock for every invocation
-from ..support.exceptions import CommandError
-from ..utils import swallow_outputs
-from ..utils import rmtemp
-from ..cmd import Runner
-from ..consts import ARCHIVES_TEMP_DIR
-from ..utils import rmtree
-from ..utils import get_tempfile_kwargs
-from ..utils import assure_unicode
-
-from ..utils import on_windows
-
-_runner = Runner()
-
-
-def _patool_run(cmd, verbosity=0, **kwargs):
-    """Decorated runner for patool so it doesn't spit out outputs to stdout"""
-    # use our runner
-    try:
-        # kwargs_ = kwargs[:];         kwargs_['shell'] = True
-        # Any debug/progress output could be spit out to stderr so let's
-        # "expect" it.
-        #
-        if isinstance(cmd, (list, tuple)) and kwargs.get('shell'):
-            # patool (as far as I see it) takes care about quoting args
-            cmd = ' '.join(cmd)
-        out, err = _runner.run(cmd,
-                    #log_stdout='offline',
-                    #log_stderr='offline',
-                    #expect_stderr=True,
-                    #stdin=open('/dev/null'),
-                    **kwargs)
-        lgr.debug("Finished running for patool. stdout=%s, stderr=%s", out, err)
-        return 0
-    except CommandError as e:
-        return e.code
-    except Exception as e:
-        lgr.error("While invoking runner caught unexpected exception: %s" % e)
-        return 100  # unknown beast
-patoolib.util.run = _patool_run
-
-
-# yoh: only keys are used atm, logic in decompress_file is replaced to use
-# patool
-
-DECOMPRESSORS = {
-    r'\.(tar\.bz|tbz)$': 'tar -xjvf %(file)s -C %(dir)s',
-    r'\.(tar\.xz)$': 'tar -xJvf %(file)s -C %(dir)s',
-    r'\.(tar\.gz|tgz)$': 'tar -xzvf %(file)s -C %(dir)s',
-    r'\.(zip)$': 'unzip %(file)s -d %(dir)s',
-}
-
-
-def unixify_path(path):
-    r"""On windows convert paths from drive:\d\file to /drive/d/file
-
-    This overcomes problems with various cmdline tools we are to use,
-    such as tar etc
-    """
-    if on_windows:
-        drive, path_ = os.path.splitdrive(path)
-        path_ = path_.split(os.sep)
-        path_ = '/'.join(path_)
-        if drive:
-            # last one must be :
-            assert(drive[-1] == ":")
-            return '/%s%s' % (drive[:-1], path_)
-        else:
-            return path_
-    else:
-        return path
+from datalad import cfg
+if cfg.get('datalad.runtime.use-patool', False):
```

Comment:
I would have also checked if `external_versions["cmd:7z"]` since if it not available (we cannot force its installation via setup.py but it is available) better to announce that (e.g. even a lgr.warning) and fall to patool which might choose another (7z is the default for many types) way to handle those files.  IMHO it would be better to just crash with "7z command is not found" message

Reply:
I will add that test.

Reply:
Done